ledger/ledger-on-memory/src/main/scala/com/daml/ledger/on/memory/InMemoryLedgerReaderWriter.scala

```diff
@@ -68,35 +75,61 @@ object InMemoryLedgerReaderWriter {
   final class SingleParticipantOwner(
       ledgerId: LedgerId,
       batchingLedgerWriterConfig: BatchingLedgerWriterConfig,
+      preExecute: Boolean,
       participantId: ParticipantId,
       timeProvider: TimeProvider = DefaultTimeProvider,
       stateValueCache: Cache[DamlStateKey, DamlStateValue] = Cache.none,
+      stateValueCacheForPreExecution: Cache[DamlStateKey, (DamlStateValue, Fingerprint)] =
+        Cache.none,
       metrics: Metrics,
       engine: Engine,
   )(implicit materializer: Materializer)
       extends ResourceOwner[KeyValueLedger] {
+
     override def acquire()(
         implicit executionContext: ExecutionContext
     ): Resource[KeyValueLedger] = {
       val state = InMemoryState.empty
       for {
         dispatcher <- dispatcherOwner.acquire()
-        readerWriter <- new Owner(
-          ledgerId,
-          batchingLedgerWriterConfig,
-          participantId,
-          metrics,
-          timeProvider,
-          stateValueCache,
-          dispatcher,
-          state,
-          engine
-        ).acquire()
+        readerWriter <- (if (preExecute)
+                           createBatchingOwner(state, dispatcher)
+                         else
+                           createPreExecutingOwner(state, dispatcher)).acquire()
       } yield readerWriter
     }
+
+    private def createPreExecutingOwner(
```

Comment:
Instead of adding separate factory functions you might want to simply add a default constructor to the below `*Owner` classes.

Reply:
The facade single-participant owner was and is actually only used by tests. Removed anyway and inlined in relevant tests.